server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -99,52 +101,49 @@ private void extensionsDiscovery() throws IOException {
         if (!FileSystemUtils.isAccessibleDirectory(extensionsPath, logger)) {
             return;
         }
-        for (final Path plugin : PluginsService.findPluginDirs(extensionsPath)) {
-            try {
-                PluginInfo pluginInfo = PluginInfo.readFromProperties(plugin);
+        List<Path> pluginDirs = PluginsService.findPluginDirs(extensionsPath);
 
-                // TODO: Add unit tests for commented section below
+        Set<Extension> extensions = new HashSet<Extension>();
+        if (!pluginDirs.isEmpty()) {
+            try {
+                extensions = readFromExtensionsYml("/config/extensions.yml").getExtensions();
+            } catch (Exception e) {
+                throw new IllegalStateException("Could not read from extensions.yml");
+            }
+        }
 
-                /**
-                try {
-                    List<Extension> extensions = readFromExtensionsYml(extensionsPath.toString()).getExtensions();
-                    for(Extension extension : extensions) {
+        for (final Path plugin : pluginDirs) {
+            try {
+                PluginInfo pluginInfo = PluginInfo.readFromProperties(plugin);
+                logger.info("Plugin Info: " + pluginInfo);
+                String pluginName = pluginInfo.getName();
+                for (Extension extension : extensions) {
+                    if (extension.getName().equals(pluginName)) {
                         extensionsSet.add(
                             new DiscoveryExtension(
                                 extension.getName(),
-                                "id",
+                                extension.getName(),
```

Comment:
Are we simply using the name as the persistent ID? Should this be a bit more unique, perhaps accounting for versioning?

Reply:
It's just a placeholder for now, but I'll try to switch it over since that makes more sense